ui/apps/platform/src/Containers/NetworkGraph/hooks/useScopeHierarchy.ts

```diff
@@ -1,24 +1,23 @@
-import useFetchClustersForPermissions from 'hooks/useFetchClustersForPermissions';
 import useURLSearch from 'hooks/useURLSearch';
+import { ClusterScopeObject } from 'services/RolesService';
 import { getScopeHierarchyFromSearch } from '../utils/hierarchyUtils';
 import { NetworkScopeHierarchy } from '../types/networkScopeHierarchy';
 
+const emptyScopeHierarchy = {
+    cluster: {
+        id: '',
+        name: '',
+    },
+    namespaces: [],
+    deployments: [],
+    remainingQuery: {},
+};
+
 /**
  * Returns the current scope hierarchy from the URL search params.
  */
-export function useScopeHierarchy(): NetworkScopeHierarchy {
+export function useScopeHierarchy(availableClusters: ClusterScopeObject[]): NetworkScopeHierarchy {
     const { searchFilter } = useURLSearch();
-    const { clusters } = useFetchClustersForPermissions(['NetworkGraph', 'Deployment']);
 
-    return (
-        getScopeHierarchyFromSearch(searchFilter, clusters) ?? {
-            cluster: {
-                id: '',
-                name: '',
-            },
-            namespaces: [],
-            deployments: [],
-            remainingQuery: {},
-        }
-    );
+    return getScopeHierarchyFromSearch(searchFilter, availableClusters) ?? emptyScopeHierarchy;
```

Comment:
I was wondering if it was necessary to have this function be abstracted out into its own module in `hierarchyUtils.ts` vs. just having the logic for `getScopeHierarchyFromSearch` laid out in this hook. Seems like a lot of jumping around to find out how the logic is supposed to work and considering how sparse this hook is in terms of lines of code, it seems a little unnecessary. I guess if that specific function was being used in other places, it makes sense to do this, but I couldn't find it being used anywhere but in this file. I know it's not part of this PR's goal so we don't have to do anything, but just thought I'd bring it up for a discussion

Reply:
That's a good point. I checked and the other exported functions in `hierarchyUtils.ts` are all also only used in a single location so it might be worth it to colocate them with their components/hooks too.